src/libaktualizr/primary/sotauptaneclient.cc

```diff
@@ -683,16 +670,20 @@ std::pair<bool, Uptane::Target> SotaUptaneClient::downloadImage(const Uptane::Ta
   return {success, target};
 }
 
-bool SotaUptaneClient::uptaneIteration(std::vector<Uptane::Target> *targets, unsigned int *ecus_count) {
-  if (!updateDirectorMeta()) {
-    LOG_ERROR << "Failed to update Director metadata: " << last_exception.what();
-    return false;
+void SotaUptaneClient::uptaneIteration(std::vector<Uptane::Target> *targets, unsigned int *ecus_count) {
+  try {
+    updateDirectorMeta();
+  } catch (const std::exception &e) {
+    LOG_ERROR << "Director metadata update failed: " << e.what();
+    throw;
```

Comment:
I'm not sure moving `try..catch` from `updateDirectorMeta` and others to the upper-level function improves readability.

Reply:
Changed in latest version